store/src/main/java/org/apache/rocketmq/store/ConsumeQueue.java

```diff
@@ -84,6 +84,58 @@ public boolean load() {
         return result;
     }
 
+    public void checkCommitLogAndConsumeQueueConsistent() {
+
+        String queueDir = this.storePath
+            + File.separator + topic
+            + File.separator + queueId + File.separator;
+
+        long lastRecordOffset = Math.max(getMaxOffsetInQueue() - 1, 0);
+
+        if (lastRecordOffset == 0) {
+            return;
+        }
+
+        SelectMappedBufferResult consumeQueueBuffer = getIndexBuffer(lastRecordOffset);
+
+        if (consumeQueueBuffer == null) {
+            String errorMsg = String.format("Can't find consume queue mappedBuffer by offset %s,queueDir %s", lastRecordOffset, queueDir);
+            log.error(errorMsg);
+            throw new IllegalStateException(errorMsg);
+        }
+
+        //position latest consume queue offset,
+        long offsetPy = consumeQueueBuffer.getByteBuffer().getLong();
+        int sizePy = consumeQueueBuffer.getByteBuffer().getInt();
+        consumeQueueBuffer.getByteBuffer().getLong();
```

Comment:
Do we need the side effect of `getLong`?

Reply:
fixed